docs/DeveloperGuide.md

```diff
@@ -1,34 +1,920 @@
+
+<br/>
+
+### 4.2 Browse Feature
+The browse feature is a useful feature that will allow users to quickly look through all 
+the different anime series available in a browsing fashion. The feature will have several enhancements such 
+as sorted browsing to browse in alphabetical or by the rating of the anime.
+
+<br/>
+
+#### 4.2.1 Current Implementation
+The `BrowseCommand` is executed by `BrowseCommandParser` after parsing the user input. It will then fetch `Anime` objects matching the parameters specified by `BrowseCommandParser` that are stored in `AnimeData`. It extends the `Command` class and implements the following operations:
+*   `BrowseCommand#sortBrowseList()` - Handles any sorting of `Anime` objects. 
+*   `BrowseCommand#buildBrowseOutput()` - Creates the output string to be printed to users.
+<br/>
+
+These are the two fundamental operations that will carry out the execution of the browse command.
+
+`BrowseCommand` will be constructed with default values. This ensures that even when there is no input provided, 
+it can still perform a default browse and return relevant useful information.
+
+Before going any further, please refer to this helpful table of the `BrowseCommand` parameters for reference.
+
+| Attribute | Option | Function          |
+| ---       | ---    | ---               |
+| order     | 0      | Ascending         |
+| order	    | 1      | Descending        |
+| sortType  | 0      | No Sort           |
+| sortType  | 1      | by name           |
+| sortType  | 2      | by rating         |
+| sortType  | 3      | back to original  |
+| page      | \>= 1  | page number       |
+
+> :bulb: The magic values have already been refactored out into constant variables.
+
+Given below is an example usage scenario to showcase how a `BrowseCommand` will behave at each step. 
+This example will utilise a small sample list of `AnimeData` and a page size of `3`.
+
+> :memo: It is set to `20` as default in actual execution. 
+
+**Step 1:** When the user enters the command for `Browse`, the input will be processed and parsed by `parser.java` and 
+then further parsed by `BrowseParser.java`.
+
+**Step 2:** Upon completion of parsing and input validation, `BrowseParser.java` will create an executable `BrowseCommand` 
+object and return it to `main`.
+
+**Step 3:** Once the `BrowseCommand` object is executed, it will firstly process the `AnimeData` by calling on its `BrowseCommand#sortBrowseList()` operation to identify the order that the list will need to be in (if any).
+
+For this case since it is a default browse operation, there is no sorting performed.
+
+**Step 4:** Now `BrowseCommand` will utilise its `BrowseCommand#buildBrowseOutput()` operation to access all 
+`Anime objects within the page window, as shown in the diagram below.
+
+![Browse Object Diagram 1](images/Browse-Default-State.png) <br/>
```

Comment:
![image](https://user-images.githubusercontent.com/57039314/97525974-1b3a2c80-19e3-11eb-9034-58622aa76c71.png)

Object diagrams do make it very simple for developers to understand the entire structure of a said component/function. From what I gathered, your team is explaining steps with respect to a given example. Would a sequence diagram be better at facilitating the flow of information since the users can read in a linear flow?